1-tictactoe/tictactoe.rb

```diff
@@ -0,0 +1,90 @@
+class Tictactoe
+
+  def play
+    # Game instructions
+    puts "Welcome Players in the TicTacToe game!
+    To hit desired position just type cell address as shown bellow."
+    positions = %w(A1 A2 A3 B1 B2 B3 C1 C2 C3)
+    print_board(positions)
+
+    # Start game
+    puts "Lets play!"
+    current_player = "X"
+    hits = new_board
+    until check_winer(hits, current_player) do
+      current_player == "X" ? current_player = "O" : current_player = "X"
+      print "Player #{current_player} turn. Type cell address: "
+      address = gets.chomp
+      valid_shot = check_position(address, hits)
+      until valid_shot do
+        print "You gave wrong address, try one more time: "
+        address = gets.chomp
+        valid_shot = check_position(address, hits)
+      end
+      update_hits(valid_shot, hits, current_player)
+      print_board(hits)
+    end
+    if board_full?(hits)
+      puts "That's a tie!"
+    else
+      puts "Game won by: Player #{current_player}!"
+    end
+  end
+
+  private
+
+  def new_board
+    Array.new(9, " ")
+  end
+
+  def print_board(hits)
+    puts "    -------------
+    | #{hits[0]} | #{hits[1]} | #{hits[2]} |
+    -------------
+    | #{hits[3]} | #{hits[4]} | #{hits[5]} |
+    -------------
+    | #{hits[6]} | #{hits[7]} | #{hits[8]} |
+    -------------"
+  end
+
+  def board_full?(board)
+    board.select{ |i| i == "X" or i == "O" }.length == 9 ? true : false
+  end
+
+  def update_hits(position, board, player)
+    board[position] = player
+  end
+
+  def check_position(address, board)
+    address.capitalize!
+    case address
+      when "A1"; position = 0
+      when "A2"; position = 1
+      when "A3"; position = 2
+      when "B1"; position = 3
+      when "B2"; position = 4
+      when "B3"; position = 5
+      when "C1"; position = 6
+      when "C2"; position = 7
+      when "C3"; position = 8
+      else false
+    end
+    if position and board[position] == " "
```

Comment:
Use `&&` instead of `and` to satisfy operator precedence https://ruby-doc.org/core-2.4.0/doc/syntax/precedence_rdoc.html